src/infi/systray/traybar.py

```diff
@@ -122,15 +122,25 @@ def shutdown(self):
         PostMessage(self._hwnd, WM_CLOSE, 0, 0)
         self._message_loop_thread.join()
 
-    def update(self, icon=None, hover_text=None):
-        """ update icon image and/or hover text """
+    def update(self, icon=None, hover_text=None, menu_options=None):  #  "menu_options=None" added to be allow the update of the menu options
+        """ update icon image and/or hover text and/or menu options"""
         if icon:
             self._icon = icon
             self._load_icon()
         if hover_text:
             self._hover_text = hover_text
+        # "if menu_options" added to be allow the update of the menu options
+        if menu_options:
+            menu_options = menu_options or ()
+            menu_options = menu_options + (('Quit', None, SysTrayIcon.QUIT),)
+            self._next_action_id = SysTrayIcon.FIRST_ID
+            self._menu_actions_by_id = set()
+            self._menu_options = self._add_ids_to_menu_options(list(menu_options))
+            self._menu_actions_by_id = dict(self._menu_actions_by_id)
+            self._menu = None  # detroy the old menu created by right clicking the icon

```

Comment:
What I'm afraid of here is that we're not freeing the previous menu (and submenus and icons, maybe?), potentially causing resource leaks.

Reply:
It's possible but I don't know how to prevent this.

Reply:
Would `DestroyMenu` do the trick?
https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-destroymenu

You'll probably have to add an alias for it in `win32_adapter.py` but I managed to get this working by doing:
```python
if menu_options:
    [the menu update code]
    ctypes.windll.user32.DestroyMenu(self._menu)
    self._menu = None
```